### PR TITLE
`sage.schemes`: Deprecate `is_...` functions

### PR DESCRIPTION
**Pull request by mkoeppe**
Also a small improvement in the script `src/bin/sage-fixdoctests` that helped in preparing the PR.

### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/schemes/affine/affine_space.py [hunk 1]
```diff
@@ -46,12 +46,17 @@ def is_AffineSpace(x) -> bool:
...
+        See https://github.com/sagemath/sage/issues/38022 for details.
         True
         sage: is_AffineSpace(AffineSpace(5, GF(9, 'alpha'), names='x'))                 # needs sage.rings.finite_rings
         True
         sage: is_AffineSpace(Spec(ZZ))
         False
     """
+    from sage.misc.superseded import deprecation
+    deprecation(38022, "the function is_AffineSpace is deprecated; use 'isinstance(..., AffineSpace_generic)' instead")
     return isinstance(x, AffineSpace_generic)
```
Comment (kwankyu) [inline src/sage/schemes/affine/affine_space.py:60]: According to the developer guide 

https://doc.sagemath.org/html/en/developer/coding_in_python.html#experimental-unstable-code

deprecation message begins with a capital letter and ends in a period.

**Comment by kwankyu** [comment]
Otherwise lgtm.

File: src/sage/schemes/affine/affine_space.py (hunk 1 above)
Comment (mkoeppe) [reply src/sage/schemes/affine/affine_space.py:60]: Thanks for the pointer; I've done this wrong in a bunch of other recent PRs, I think.

Fixed in ceebb70482a

Comment (kwankyu) [reply src/sage/schemes/affine/affine_space.py:60]: Perhaps I also have missed it before. This time I could see the style distinguished from that of the surrounding text.

Thanks.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**